community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/Predicate.scala

```diff
@@ -96,6 +96,53 @@ case class Ands(predicates: List[Predicate]) extends Predicate {
   override def atoms: Seq[Predicate] = predicates
 }
 
+case class AndedPropertyComparablePredicates(ident: Identifier, prop: Property, comparables: NonEmptyList[ComparablePredicate]) extends Predicate {
+
+  def symbolTableDependencies: Set[String] = comparables.foldLeft(prop.symbolTableDependencies) {
+    case (acc, predicate) => acc ++ predicate.symbolTableDependencies
+  }
+
+  def containsIsNull: Boolean = comparables.exists(_.containsIsNull)
+
+  def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = {
```

Comment:
This code was simply copied from above, lines 70-90. I would like to not touch it.

One idea would be to let this class `AndedPropertyComparablePredicates` extend `Ands` instead of being a sibling-copy, but there's a disadvantage to that in that instances may then match on pattern-matches using `Ands`, and we do not want that. However, I am unsure on how plausible this disadvantage is, ie how often such a match would actually occur.